pkg/notes/notes_data.go

```diff
@@ -513,3 +532,74 @@ func setEmptyComponentGitHubRepositoryFromOtherKilnfile(k1, k2 cargo.Kilnfile) c
 }
 
 func closeAndIgnoreError(c io.Closer) { _ = c.Close() }
+
+type TrainstatFetcher interface {
+	FetchTrainstatNotes(milestone string, version string, tile string) ([]string, error)
+}
+
+type TrainstatClient struct {
+	host       string
+	httpClient *http.Client
+}
+
+func NewTrainstatClient(host string) TrainstatClient {
+	if host == "null" || strings.TrimSpace(host) == "" {
+		host = TrainstatUrl()
+	}
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return TrainstatClient{
+		host:       host,
+		httpClient: client,
+	}
+}
+
+func (t *TrainstatClient) FetchTrainstatNotes(milestone string, version string, tile string) (notes []string, err error) {
+	if !t.tileSupported(tile) {
+		return []string{}, nil
+	}
+
+	baseURL := fmt.Sprintf("%s/%s", t.host, "/api/v1/release_notes")
+	req, err := http.NewRequest(http.MethodGet, baseURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	q := req.URL.Query()
+	q.Set("milestone", milestone)
+	q.Set("version", version)
+	q.Set("tile", tile)
+	req.URL.RawQuery = q.Encode()
+
+	res, err := t.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	responseData, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(responseData, &notes); err != nil {
+		return notes, err
+	}
+	return
+}
+
+func (t *TrainstatClient) tileSupported(tile string) bool {
+	switch tile {
+	case "elastic-runtime":
```

Comment:
consider multi value switch statement.

```go
switch tile {
case "a", "b", "c":
  return true
default:
  return false
}
```

Reply:
I would prefer not to have the slugs in the code and for the code to handle a "404" from train stat properly.

Reply:
Thanks @crhntr , didn't know we could use switch statements like that.

The purpose of this function is to know if we should call trainstat to fetch notes. Trainstat only supports TAS tiles, and for the acceptance tests we are running the fetch release notes logic for the hello-tile.